src/modules/meals.js

```diff
@@ -26,22 +30,51 @@ export const fetchMealsByCategory = async (cat) => {
         <h3>${item.strMeal}</h3>
         <div class="meal-info">
           <div class="comments">
-            <i class="bi bi-chat"></i>
+            <i class="bi bi-chat" id="comment-${item.idMeal}"></i>
             <p>50</p>
           </div>
           <div class="likes">
-            <i class="bi bi-heart"></i>
-            <p>${json1.find((x) => x.item_id === item.idMeal) ? json1.find((x) => x.item_id === item.idMeal).item_id : '0'}</p>
+            <div class="like-button">
+              <div class="heart-bg heart-bg-hover">
+                <div class="heart-icon" id="like-${item.idMeal}"></div>
+              </div>
+              <div class="likes-amount" id="like-amount-${item.idMeal}">${json1.find((x) => x.item_id === item.idMeal) ? json1.find((x) => x.item_id === item.idMeal).likes : '0'}</div>
+            </div>
           </div>
         </div>
       `);
         mealElem.firstElementChild.id = `meal-${item.idMeal}`;
         meals.appendChild(mealElem);
 
+        if (localStorage.getItem('likes')) {
+          if (JSON.parse(localStorage.getItem('likes')).includes(item.idMeal)) {
+            document.getElementById(`like-${item.idMeal}`).classList.toggle('liked');
+            document.getElementById(`like-${item.idMeal}`).parentNode.classList.toggle('heart-bg-hover');
+            document.getElementById(`like-amount-${item.idMeal}`).style.color = '#f91880';
+            document.getElementById(`like-amount-${item.idMeal}`).textContent = json1.find((x) => x.item_id === item.idMeal).likes;
+          }
+        }
+
         document.getElementById(`meal-${item.idMeal}`).addEventListener('click', () => {
           const popup = new Popup();
           popup.createPopup(item.idMeal);
         });
+
+        document.getElementById(`comment-${item.idMeal}`).addEventListener('click', () => {
+          showComments('item1');
+        });
```

Comment:
kindly use item.idMeal as a Parameter to showComments instead of 'item1' so each post will have its own comments